tests/components/bryant_evolution/test_climate.py

```diff
@@ -0,0 +1,348 @@
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+
+async def test_setup_integration_success(
+    hass: HomeAssistant, mock_evolution_entry: MockConfigEntry
+) -> None:
+    """Test that an instance can be constructed."""
+    state = hass.states.get("climate.system_1_zone_1")
+    assert state, (x.name() for x in hass.states.async_all())
+    assert state.state == "cool"
+    assert state.attributes["fan_mode"] == "auto"
+    assert state.attributes["current_temperature"] == 75
+    assert state.attributes["temperature"] == 72
+
+
+async def test_setup_integration_prevented_by_unavailable_client(
+    hass: HomeAssistant, mock_evolution_client_factory: AsyncMock
+) -> None:
+    """Test that setup throws ConfigEntryNotReady when the client is unavailable."""
+    mock_evolution_client_factory.side_effect = FileNotFoundError("test error")
+    mock_evolution_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_FILENAME: "test_setup_integration_prevented_by_unavailable_client",
+            CONF_SYSTEM_ZONE: [(1, 1)],
+        },
+    )
+    mock_evolution_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_evolution_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_evolution_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_setup_integration_client_returns_none(
+    hass: HomeAssistant, mock_evolution_client_factory: AsyncMock
+) -> None:
+    """Test that an unavailable client causes ConfigEntryNotReady."""
+    mock_client = AsyncMock(spec=BryantEvolutionLocalClient)
+    mock_evolution_client_factory.side_effect = None
+    mock_evolution_client_factory.return_value = mock_client
+    mock_client.read_fan_mode.return_value = None
+    mock_client.read_current_temperature.return_value = None
+    mock_client.read_hvac_mode.return_value = None
+    mock_client.read_cooling_setpoint.return_value = None
+    mock_client.read_zone_name.return_value = None
+    mock_evolution_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_FILENAME: "/dev/ttyUSB0", CONF_SYSTEM_ZONE: [(1, 1)]},
+    )
+    mock_evolution_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_evolution_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_evolution_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_setup_multiple_systems_zones(
+    hass: HomeAssistant,
+    mock_evolution_client_factory: AsyncMock,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test that a device with multiple systems and zones works."""
+    hass.config.units = US_CUSTOMARY_SYSTEM
+    mock_evolution_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_FILENAME: "/dev/ttyUSB0", CONF_SYSTEM_ZONE: DEFAULT_SYSTEM_ZONES},
+    )
+    mock_evolution_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_evolution_entry.entry_id)
+    await hass.async_block_till_done()
```

Comment:
Why do we have `mock_evolution_entry` if we are not using it?

Reply:
sorry, aren't we using it? eg line 122

Reply:
we also want the side-effects of setting it up (line 117)

Reply:
Oh I missed that then. But I thought this looked a lot like the fixture